src/gt4py/dace_lazy_stencil.py

```diff
@@ -0,0 +1,63 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence, Set, Tuple
```

Comment:
I like that the `DaceStencilObject` is stored in the `backend` module. Can this be moved there too?